packages/cadl-ranch-specs/http/type/model/inheritance/enum-discriminator/main.tsp

```diff
@@ -0,0 +1,50 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+
+@doc("Illustrates inheritance with enum discriminator.")
+@scenarioService("/type/model/inheritance/enum-discriminator")
+namespace Type.Model.Inheritance.EnumDiscriminator;
+
+@doc("extensible enum type for discriminator")
+enum DogKind {
+  Golden: "golden",
+}
+
+@doc("Test extensible enum type for discriminator")
+@discriminator("kind")
+model Dog {
+  kind: DogKind;
+  weight: int32;
+}
+
+model Golden extends Dog {
+  kind: DogKind.Golden;
+}
+
+@scenario
+@scenarioDoc("""
+Receive model with extensible enum discriminator type.
+Expected response body:
+```json
+
+{"kind": "golden", "weight": 10}
+```
+""")
+@route("/extensible-enum")
+@get
+op getModel(): Dog;
+
+@scenario
+@scenarioDoc("""
+Send model with extensible enum discriminator type.
+Expected request body:
+```json
+
+{"kind": "golden", "weight": 10}
+```
+""")
+@route("/extensible-enum")
+@put
+op putModel(@body input: Dog): OkResponse;
```

Comment:
Should we have the same scenarios as for the non enum tests?
- get
- put
- get unknown discriminator
- get missing discriminator property? (not sure this one is really something that can happen so would be ok not having it in either too)

Reply:
added `getMissingDiscriminator` and `getWrongDiscriminator` cases